interfaces/builtin/desktop_launch.go

```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopLaunchSummary = `allows snaps to identify and launch other snaps`
+
+const desktopLaunchBaseDeclarationPlugs = `
+  desktop-launch:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const desktopLaunchBaseDeclarationSlots = `
+  desktop-launch:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const desktopLaunchConnectedPlugAppArmor = `
+# Description: Can identify and launch other snaps.
+
+# Access to the desktop files installed by snaps
+/var/lib/snapd/desktop/applications/{,*} r,
+
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/io/snapcraft/Launcher
+    interface=io.snapcraft.Launcher
+    member=OpenDesktopEntryEnv
+    peer=(label=unconfined),
+`
+
+// Only implicitOnClassic since userd isn't yet usable on core
+func init() {
+	registerIface(&commonInterface{
+		name:                  "desktop-launch",
+		summary:               desktopLaunchSummary,
+		implicitOnClassic:     true,
+		baseDeclarationPlugs:  desktopLaunchBaseDeclarationPlugs,
```

Comment:
@jdstrand I've just got back to this. Am I missing a trick? Or do I really have to comment this out in order to install the confined-shell snap for testing?

Reply:
If you do an unasserted install with `--dangerous`, it should be installable, but the interface will not be connected. If confined-shell doesn't have a snap declaration to allow installation, it cannot be installed via the store. I just checked and confined-shell did not have a snap declaration, so I just added it the for now allows installation but requires manual connection.

Reply:
Thanks, that's done it